internal/tls/config.go

```diff
@@ -59,7 +59,7 @@ func NewServerConfig(logger log.Logger, certFile, keyFile, clientCAFile, minVers
 
 		certPool := x509.NewCertPool()
 		if !certPool.AppendCertsFromPEM(caPEM) {
-			return nil, fmt.Errorf("building client CA: %w", err)
+			return nil, errors.New("building client CA failed")
```

Comment:
nit rather than `building`, since we are not generating a CA, consider `parsing`